### PR TITLE
check_cpu_utilization: cpu utilization check with all hv flags

### PR DESCRIPTION
**Pull request by kimi1978wy**
Check host cpu utilization for each vcpu is less than 5% when
guest is idle with all hv flags

id: 1821603

Signed-off-by: Yu Wang <wyu@redhat.com>

### CODE REVIEW

**Comment by kimi1978wy** [comment]
(01/21) repeat1.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2974.00 s)
 (02/21) repeat1.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2986.57 s)
 (03/21) repeat1.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2867.04 s)
 (04/21) repeat1.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2885.35 s)
 (05/21) repeat1.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2892.09 s)
 (06/21) repeat1.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.i386.1.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2857.45 s)
 (07/21) repeat1.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.x86_64.1.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2877.56 s)
 (08/21) repeat2.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2976.99 s)
 (09/21) repeat2.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2991.82 s)
 (10/21) repeat2.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2869.01 s)
 (11/21) repeat2.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2885.06 s)
 (12/21) repeat2.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2891.57 s)
 (13/21) repeat2.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.i386.1.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2858.58 s)
 (14/21) repeat2.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.x86_64.1.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2870.63 s)
 (15/21) repeat3.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2977.30 s)
 (16/21) repeat3.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2990.30 s)
 (17/21) repeat3.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2868.26 s)
 (18/21) repeat3.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2889.47 s)
 (19/21) repeat3.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2892.46 s)
 (20/21) repeat3.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.i386.1.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2856.48 s)
 (21/21)  repeat3.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.x86_64.1.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2871.23 s)

**Comment by kimi1978wy** [comment]
Could you help to review this patch? @YongxueHong 

 (01/14) repeat1.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (3006.45 s)
 (02/14) repeat1.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (3001.91 s)
 (03/14) repeat1.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2877.05 s)
 (04/14) repeat1.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2884.50 s)
 (05/14) repeat1.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2895.18 s)
 (06/14) repeat1.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.i386.1.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2865.90 s)
 (07/14) repeat1.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.x86_64.1.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2878.62 s)
 (08/14) repeat2.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (3005.97 s)
 (09/14) repeat2.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2996.56 s)
 (10/14) repeat2.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2875.04 s)
 (11/14) repeat2.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2885.64 s)
 (12/14) repeat2.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2905.96 s)
 (13/14) repeat2.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.i386.1.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2865.73 s)
 (14/14) repeat2.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.x86_64.1.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2876.39 s)

File: qemu/tests/check_cpu_utilization.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        if not service_item:
+            return
+
+        s, o = session.cmd_status_output(service_check_cmd % service)
+        if s:
+            test.error("Failed to get status for service: %s, "
+                       "status=%s, output=%s" % (service, s, o))
+        if re.search(r'STOPPED', o, re.I | re.M):
+            return
+        session.cmd(service_stop_cmd % (service, service))
```
Comment (YongxueHong) [inline qemu/tests/check_cpu_utilization.py:89]: Suggest that put  `_stop_service`  on the above `run(test, params, env)`.

File: qemu/tests/cfg/check_cpu_utilization.cfg [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    master_images_clone = image1
+    remove_image_image1 =yes
+    nics = ""
+    cdroms = ""
+    cpu_model_flags = "hv_crash"
+    host_check_times = 900
+    host_check_interval = 2
+    serives_to_stop = wuauserv bits dosvc SysMain
+    service_check_cmd = sc query %s
+    service_stop_cmd = sc stop %s & sc config %s start=disabled
```
Comment (YongxueHong) [inline qemu/tests/cfg/check_cpu_utilization.cfg:14]: Better `service_stop_cmd = sc stop {0} & sc config {0} start=disabled`

File: qemu/tests/check_cpu_utilization.py (hunk 1 above)
Comment (YongxueHong) [inline qemu/tests/check_cpu_utilization.py:89]: Better `session.cmd(service_stop_cmd.format(service))`

**Comment by kimi1978wy** [comment]
(1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.i386.1.io-github-autotest-qemu.check_cpu_utilization.q35: PASS (2846.36 s)

**Pull request merged**